io/disk/disktest.py

```diff
@@ -0,0 +1,115 @@
+from avocado import main
+from avocado.utils import memory
+from avocado.utils import disk
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Disktest(Test):
+
+    """
+    Avocado module for disktest.
+    Pattern test of the disk, using unique signatures for each block and each
+    iteration of the test. Designed to check for data corruption issues in the
+    disk and disk controller.
+    It writes 50MB/s of 500KB size ops.
+    """
+    version = 2
+
+    def setUp(self):
+        """
+        Verifies if we have gcc to compile disktest.
+        """
+        softm = SoftwareManager()
+        if not softm.check_installed("gcc") and not softm.install("gcc"):
+            self.error('Gcc is needed for the test to be run')
+        self.disks = self.params.get('disks', default=None)
+        self.gigabytes = self.params.get('gigabytes', default=None)
+        self.chunk_mb = self.params.get('chunk_mb', default=None)
+        self.memory_mb = memory.memtotal() / 1024 / 8
+        self.data_dir = os.path.join(os.path.abspath(self.datadir))
+        self.disk_srcdir = os.path.join(os.path.abspath(self.datadir), 'src')
+        self.build(self.disk_srcdir)
+
+    def build(self, srcdir):
+        """
+        Compiles disktest.
+        """
+        os.chdir(srcdir)
+        build.make(srcdir, extra_args='clean')
+        build.make(srcdir)
+
+    def one_disk_chunk(self, disks, chunk):
+        """
+        Tests one part of the disk by spawning a disktest instance.
+        :param disks: Directory (usually a mountpoint).
+        :param chunk: Portion of the disk used.
+        """
+        logging.info("Testing %d MB files on %s in %d MB memory, chunk %s",
+                     self.chunk_mb, disks, self.memory_mb, chunk)
+        logfile = self.outputdir + '/log.txt'
+        cmd = ("%s/disktest -m %d -f %s/testfile.%d -i -S >>%s 2>&1" %
+               (self.disk_srcdir, self.chunk_mb, disks, chunk, logfile))
+        logging.debug("Running '%s'", cmd)
+        thread = subprocess.Popen(cmd, shell=True)
+        return thread.pid
+
+    def cleanup(self, disks):
+        """
+        To clean all the testfiles generated
+        """
+        process.run('rm -rf %s/testfile.*' % disks, shell=True)
+
+    def test(self):
+        """
+        Runs one iteration of disktest.
+        :param disks: Directory (usually mountpoints) to be passed
+                to the test. Directory gets created if not existing
+        :param gigabytes: Disk space that will be used for the test to run.
+        :param chunk_mb: Size of the portion of the disk used to run the test.
+                Cannot be larger than the total amount of free RAM.
+        """
+        os.chdir(self.disk_srcdir)
+        if self.chunk_mb is None:
+            self.chunk_mb = memory.memtotal() / 1024 / 8
+        if self.disks is None:
+            self.disks = self.data_dir
```

Comment:
this location might not be writable. You should probably use `self.workdir` and as `srcdir` is inside of this, I'd recommend using `avocado.utils.path.init_dir(self.workdir, "testdisk")` to avoid possible problems.